VideoAPI/VideoApi.UnitTests/Controllers/Conference/CloseConferenceTests.cs

```diff
@@ -56,15 +60,47 @@ public async Task Should_close_conference_and_delete_audio_recording_application
             AudioPlatformServiceMock.Setup(v => v.DeleteAudioApplicationAsync(It.IsAny<Guid>())).ReturnsAsync(response);
             AzureStorageServiceFactoryMock.Setup(x => x.Create(AzureStorageServiceType.Vh)).Returns(AzureStorageServiceMock.Object);
 
-            AzureStorageServiceMock.Setup(x => x.FileExistsAsync(It.IsAny<string>())).ReturnsAsync(true);
 
+            _blobClientMock = new Mock<BlobClient>();
+            _blobClientMock.Setup(x => x.Name).Returns("SomeBlob");
+            AzureStorageServiceMock.Setup(x => x.GetAllBlobsAsync(It.IsAny<string>())).Returns(GetMockBlobClients);
+
+            await Controller.CloseConferenceAsync(Guid.NewGuid());
+
+            CommandHandlerMock.Verify(c => c.Handle(It.IsAny<CloseConferenceCommand>()), Times.Once);
+            AudioPlatformServiceMock.Verify(v => v.DeleteAudioApplicationAsync(It.IsAny<Guid>()), Times.Once);
+        }
+
+        [Test]
+        public async Task Should_close_conference_and_delete_audio_recording_application_if_audio_files_exist_and_actual_start_date_is_null()
+        {
+            TestConference.AudioRecordingRequired = true;
+            QueryHandlerMock
+              .Setup(x => x.Handle<GetConferenceByIdQuery, VideoApi.Domain.Conference>(It.IsAny<GetConferenceByIdQuery>()))
+              .ReturnsAsync(TestConference);
+            var response = new AudioPlatformServiceResponse(true);
+            AudioPlatformServiceMock.Setup(v => v.DeleteAudioApplicationAsync(It.IsAny<Guid>())).ReturnsAsync(response);
+            AzureStorageServiceFactoryMock.Setup(x => x.Create(AzureStorageServiceType.Vh)).Returns(AzureStorageServiceMock.Object);
+
+
+            _blobClientMock = new Mock<BlobClient>();
+            _blobClientMock.Setup(x => x.Name).Returns($"{TestConference.HearingRefId.ToString()}.mp4");
+            AzureStorageServiceMock.Setup(x => x.GetAllBlobsAsync(It.IsAny<string>())).Returns(GetMockBlobClients);
 
             await Controller.CloseConferenceAsync(Guid.NewGuid());
 
             CommandHandlerMock.Verify(c => c.Handle(It.IsAny<CloseConferenceCommand>()), Times.Once);
             AudioPlatformServiceMock.Verify(v => v.DeleteAudioApplicationAsync(It.IsAny<Guid>()), Times.Once);
         }
 
+
+        private async IAsyncEnumerable<BlobClient> GetMockBlobClients()
+        {
+            yield return _blobClientMock.Object;
+
+            await Task.CompletedTask;
```

Comment:
maybe we can do this a little differenly

Reply:
i should have included suggestions..
`IAsyncEnumerable` requires `async` and an async without an await will give a warning so i see why its added.. as its test maybe can ignore..